addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/ExtensionWappalyzer.java

```diff
@@ -200,6 +204,24 @@ boolean isWappalyzerEnabled() {
         return enabled;
     }
 
+    void setExhaustive(boolean exhaustive) {
+        if (this.exhaustive == exhaustive) {
+            return;
+        }
+        this.exhaustive = exhaustive;
+
+        wappalyzerParam.setExhaustive(exhaustive);
+        getPassiveScanner().setExhaustive(exhaustive);
+
+        if (hasView()) {
+            getTechPanel().getExhaustiveToggleButton().setSelected(exhaustive);
```

Comment:
Should be done in the EDT.

Reply:
For the enabled button too?

Reply:
You need to execute in the EDT (e.g. `ThreadUtils.invokeLater`).